### ISSUE TITLE
cancel() does not stop immediately

### ISSUE THREAD

**Issue by lyonlim** [tra/periodic#1]
Hi,

First off, thanks for this really great implementation. I'm currently trying to cancel the whole polling if the server script returns an error. How I do this is the error message is communicated via json from the server. My code is like this:

success: function(data) {
  if (data.status == 0) {
    alert(data.msg);
    this.periodic.cancel();
  }
}

the problem with the above is that after the first error is received, periodic goes on to allow another cycle before it finally stops completely. The user in this instance will receive two alert boxes.

How can I make it such that it stops the moment i call cancel()?

Thanks :)

**Comment by lyonlim** [comment]
What I did....think it's something of a hack..

I added a new function stop() and changed periodic.defaults to use stop() instead of cancel().

  function stop() {
    stop = true;
    clearTimeout(settings.tid);
  }

declared a new var
var stop = false; (after this line: var prev_ajax_response = '';)

then added this line in run()
if (stop == true) { return false; }

Seems to work. :) Let me know if this is a proper solution.
